modules/pipeline/services/pipelinesvc/secret.go

```diff
@@ -183,26 +183,25 @@ func (s *PipelineSvc) FetchSecrets(p *spec.Pipeline) (secrets, cmsDiceFiles map[
 		}
 	}
 
-	for _, ns := range namespaces {
-		configs, err := s.cmsService.GetCmsNsConfigs(apis.WithInternalClientContext(context.Background(), "pipeline"),
-			&pb.CmsNsConfigsGetRequest{
-				Ns:             ns,
-				PipelineSource: p.PipelineSource.String(),
-				GlobalDecrypt:  true,
-			})
-		if err != nil {
-			return nil, nil, nil, nil, err
+	batchConfigs, err := s.cmsService.BatchGetCmsNsConfigs(apis.WithInternalClientContext(context.Background(), "pipeline"),
```

Comment:
I think you should dig out the specific time slice of 300ms about.
if the limitation is decryption, `batch-get` should have same time cost as get-one-by-one (after I saw the code of `batch-get`).

Reply:
Most of the time is spent on encryption，a decryption takes about 10ms。
And this optimization see pr:https://github.com/erda-project/erda/pull/4585